frontends/common/constantFolding.cpp

```diff
@@ -37,7 +62,7 @@ const IR::Expression* DoConstantFolding::getConstant(const IR::Expression* expr)
         // Casts of a constant to a value with type Type_Newtype
         // are constants, but we cannot fold them.
         if (getConstant(cast->expr))
-            return expr;
+            return CloneConstants::clone(expr);
```

Comment:
If we haven't succeeded in folding the constant, why does it still need to be cloned?

Reply:
We cannot replace these expressions with constant values, we have to leave them as (cast)constant. But we still have to clone the constant itself - due to the same reasons that we clone constants everywhere in this pass. (The cast will disappear after the `type` constructs are eliminated.) This was the case that was triggering the reported bug.